### PR TITLE
chore: add assertion for snapshot generator

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._

- Fixes #[Link the issue here]

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14439)

### CODE REVIEW

**Comment by sundy-li** [comment]
This must be the bug of `field_leaf_default_values`, it's missing to deal with `Scalar::Map` and `Scalar::Array`.  cc @lichuang

File: src/query/expression/src/schema.rs [hunk 1]
```diff
@@ -557,6 +557,7 @@ impl TableSchema {
                         )
                     });
                 }
+                Scalar::Map(_) | Scalar::Array(_) => {}
                 _ => {
```
Comment (BohuTANG) [inline src/query/expression/src/schema.rs:561]: Prefer to remove '_,' give all the case, we will never missed anything.

Comment (sundy-li) [reply src/query/expression/src/schema.rs:561]: Scalar is a long enum, added `debug_assert!(!default_value.is_nested_scalar());`

**Comment by andylokandy** [comment]
Merged to https://github.com/datafuselabs/databend/pull/14432

**Pull request closed without merging**